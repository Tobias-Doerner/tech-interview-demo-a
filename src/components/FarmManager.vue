<template>
  <div>
    <div class="text-center mb-4">
      <span class="text-3xl">Farm Manager</span>
    </div>
    <animal-counter
      v-model="animal.count"
      v-for="animal in farm.animals"
      :key="animal.species"
      :species="animal.species"
    />
  </div>
</template>

<script>
import AnimalCounter from '@/components/AnimalCounter'

export default {
  components: {
    AnimalCounter
  },
  data() {
    return {
      farm: {
        animals: [
          {
            count: 0,
            species: 'cow'
          },
          {
            count: 0,
            species: 'pig'
          },
          {
            count: 0,
            species: 'sheep'
          }
        ]
      }
    }
  }
}
</script>
