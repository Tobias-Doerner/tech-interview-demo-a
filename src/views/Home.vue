<template>
  <farm-manager />
</template>

<script>
import FarmManager from '@/components/FarmManager.vue'

export default {
  components: {
    FarmManager
  }
}
</script>
