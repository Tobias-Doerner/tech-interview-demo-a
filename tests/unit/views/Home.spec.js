import { mount, shallowMount } from '@vue/test-utils'
import axios from 'axios'
import FarmManager from '@/components/FarmManager'
import Home from '@/views/Home'

jest.mock('axios')

describe('Home', () => {
  describe('test', () => {
    test('should contain FarmManager component', () => {
      const wrapper = shallowMount(Home)
      const farmManager = wrapper.findComponent(FarmManager)
      expect(farmManager.exists()).toBe(true)
    })
  })

  describe('snapshot', () => {
    test('should render view correctly', async () => {
      axios.get.mockResolvedValue({
        status: 200,
        data: {
          animals: [
            {
              count: 1,
              species: 'cow'
            },
            {
              count: 2,
              species: 'pig'
            },
            {
              count: 3,
              species: 'sheep'
            }
          ],
          name: 'Meierhof'
        }
      })

      const wrapper = mount(Home)
      const farmManager = wrapper.findComponent(FarmManager)
      farmManager.vm.farm.name = 'Meierhof'
      farmManager.vm.load()

      await wrapper.vm.$nextTick()
      await new Promise((resolve) => setTimeout(resolve))

      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
