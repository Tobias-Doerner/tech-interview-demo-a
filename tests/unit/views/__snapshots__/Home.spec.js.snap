// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot should render view correctly 1`] = `
<div>
  <div
    class="text-center mb-4"
  >
    <span
      class="text-3xl"
    >
      Farm Manager
    </span>
  </div>
   
  <div
    class="flex justify-center items-end"
  >
    <div
      class="max-w-xs mx-2"
    >
      <label
        class="textfield__label"
      >
        
    Farm
  
      </label>
       
      <input
        class="textfield__input"
        placeholder="(e.g.: Meierhof)"
        type="text"
      />
    </div>
     
    <div>
      <button
        class="base flex items-center primary"
      >
        Load 
        <!---->
      </button>
    </div>
  </div>
   
  <div
    class="m-4"
  >
    <div
      class="flex justify-center items-center"
    >
      <button
        class="base flex items-center m-1 secondary"
      >
        
    Remove
   
        <!---->
      </button>
       
      <span
        class="w-52 text-center"
      >
        Current cow count: 1
      </span>
       
      <button
        class="base flex items-center secondary"
      >
        Add 
        <!---->
      </button>
    </div>
    <div
      class="flex justify-center items-center"
    >
      <button
        class="base flex items-center m-1 secondary"
      >
        
    Remove
   
        <!---->
      </button>
       
      <span
        class="w-52 text-center"
      >
        Current pig count: 2
      </span>
       
      <button
        class="base flex items-center secondary"
      >
        Add 
        <!---->
      </button>
    </div>
    <div
      class="flex justify-center items-center"
    >
      <button
        class="base flex items-center m-1 secondary"
      >
        
    Remove
   
        <!---->
      </button>
       
      <span
        class="w-52 text-center"
      >
        Current sheep count: 3
      </span>
       
      <button
        class="base flex items-center secondary"
      >
        Add 
        <!---->
      </button>
    </div>
  </div>
   
  <div
    class="flex justify-center"
  >
    <button
      class="base flex items-center primary"
    >
      Submit 
      <!---->
    </button>
  </div>
</div>
`;
